chainer/functions/array/spatial_transformer_grid.py

```diff
@@ -16,14 +16,15 @@ class SpatialTransformerGrid(function.Function):
 
     def __init__(self, output_shape):
         self.output_shape = output_shape
+        self._dtype = chainer.get_dtype()
```

Comment:
I overlooked it; I think it's better to accept `theta` of any dtype with kind `f` like many other functions do instead of restricting it to the current dtype.

Reply:
Fixed!